eclipse-collections-code-generator/src/main/java/org/eclipse/collections/codegenerator/EclipseCollectionsCodeGenerator.java

```diff
@@ -16,37 +16,36 @@
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.List;
+import java.util.Objects;
 import java.util.zip.CRC32;
 
 import org.eclipse.collections.codegenerator.model.Primitive;
 import org.eclipse.collections.codegenerator.tools.FileUtils;
 import org.eclipse.collections.codegenerator.tools.IntegerOrStringRenderer;
 import org.stringtemplate.v4.ST;
-import org.stringtemplate.v4.STErrorListener;
 import org.stringtemplate.v4.STGroupFile;
-import org.stringtemplate.v4.misc.STMessage;
 
 public class EclipseCollectionsCodeGenerator
 {
-    public static final String GENERATED_TEST_SOURCES_LOCATION = "target/generated-test-sources/java/";
-    public static final String GENERATED_SOURCES_LOCATION = "target/generated-sources/java/";
-    public static final String GENERATED_SERVICES_LOCATION = "target/generated-sources/resources/META-INF/services";
     private final String templateDirectory;
-    private final File moduleBaseDir;
     private final List<URL> classPathURLs;
-    private boolean isTest;
-    private boolean isService;
-    private STGroupFile templateFile;
-    private final STErrorListener stErrorListener;
-    private URL url;
+    private final ErrorListener errorListener;
+    private final File outputDirectory;
+    private final String fileExtension;
+
     private int numFileWritten = 0;
 
-    public EclipseCollectionsCodeGenerator(String templateDirectory, File moduleBaseDir, List<URL> classPathURLs, ErrorListener errorListener)
+    public EclipseCollectionsCodeGenerator(
+            String templateDirectory,
+            List<URL> classPathURLs,
+            ErrorListener errorListener,
+            File outputDirectory, String fileExtension)
```

Comment:
Move `fileExtension` to a new line.

Reply:
Fixed.

Reply:
After this lands, I have work in progress to reduce duplication here eb785181b346de9658a88315a4b2e82e065be980